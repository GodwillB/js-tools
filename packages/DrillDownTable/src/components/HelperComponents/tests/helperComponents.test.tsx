import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { DropDownCell, NullDataComponent, Spinner } from '../index';

describe('src/components/Table/DrillDownTable/components/HelperComponents', () => {
  it('renders correctly when with children', () => {
    const props = {
      cellValue: <div>Cell value</div>,
      hasChildren: true
    };
    const wrapper = mount(<DropDownCell {...props} />);
    expect(wrapper.text()).toMatchInlineSnapshot(`"Cell value ▼"`);
  });

  it('renders correctly when without children', () => {
    const props = {
      cellValue: <div>Cell value</div>,
      hasChildren: false
    };
    const wrapper = mount(<DropDownCell {...props} />);
    expect(wrapper.text()).toMatchInlineSnapshot(`"Cell value"`);
  });

  /** tests for null data components */
  it('renders nullData components correctly', () => {
    const wrapper = mount(<NullDataComponent />);
    expect(wrapper.text()).toMatchInlineSnapshot(`"No Data Found"`);
  });

  /** tets for ripple */
  it('renders default Spinner without crashing', () => {
    const wrapper = shallow(<Spinner />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
