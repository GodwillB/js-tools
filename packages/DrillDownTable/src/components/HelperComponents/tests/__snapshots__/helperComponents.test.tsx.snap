// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/Table/DrillDownTable/components/HelperComponents renders default Spinner without crashing 1`] = `
<div
  className="spinner-border m-5"
  role="status"
>
  <span
    className="sr-only"
  >
    Loading
    ...
  </span>
</div>
`;
